internal/catalog/catalog.go

```diff
@@ -166,16 +168,77 @@ func untar(dst, version string, tektonResources map[string]contract.TektonResour
 			// to wait until all operations have completed.
 			f.Close()
 
-			if filename != "README.md" {
-				if tektonResource.Checksum != sum {
-					fmt.Fprintf(os.Stderr, "%s checksum is different than the specified checksum in the catalog file: %s", sum, tektonResource.Checksum)
-					// FIXME: maybe handle *all* file before erroring out ?
-					return fmt.Errorf("invalid checksum for %s: %s != %s", filename, sum, tektonResource.Checksum)
-				}
-				fmt.Fprintf(os.Stderr, "✅ %s\n", tektonResource.Filename)
+					if filename != "README.md" {
+							if tektonResource.Checksum != sum {
+									fmt.Fprintf(os.Stderr, "%s checksum is different than the specified checksum in the catalog file: %s", sum, tektonResource.Checksum)
+									// FIXME: maybe handle *all* file before erroring out ?
+									return fmt.Errorf("invalid checksum for %s: %s != %s", filename, sum, tektonResource.Checksum)
+							}
+							fmt.Fprintf(os.Stderr, "✅ %s\n", tektonResource.Filename)
+					}
+
+					// Add "source" annotation to task YAML file
+					if strings.HasSuffix(target, ".yaml") {
+							if err := addSourceAnnotationToTask(target, resourcesURI); err != nil {
+									return err
+							}
+					}
 			}
-		}
 	}
+	
+}
+
+func addSourceAnnotationToTask(file, resourcesURI string) error {
+	// Read the Task YAML file
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+			return err
+	}
+
+	// Parse YAML data
+	var task map[string]interface{}
+	if err := yaml.Unmarshal(data, &task); err != nil {
+			return err
+	}
+
+	// Add or update the "source" annotation
+	metadata, ok := task["metadata"].(map[interface{}]interface{})
+	if !ok {
+			metadata = make(map[interface{}]interface{})
+	}
+	annotations, ok := metadata["annotations"].(map[interface{}]interface{})
+	if !ok {
+			annotations = make(map[interface{}]interface{})
+	}
+
+	repoURL := extractRepositoryURL(resourcesURI)
+	annotations["source"] = repoURL
```

Comment:
Instead of just `source` can we add `tekton.dev/source` in order to sync with other annotations ?

Reply:
Made all the changes as per your suggestion @savitaashture